### DISCUSSION TITLE
setting value of a boolean global parameter

### DISCUSSION THREAD

**Discussion by cscoon** [Azure-Player/azure.datafactory.tools#202] [General]
I have a global parameter that is defined as a boolean.  I want to set the value to true or false during the deployment.

I already have a .csv staging file I'm using to set properties, so I add my global property there:

`factory,usmcmt0-mock9e65b-datafactory1,globalParameters.CopyData.value,$($env:CopyData)`

Note that I'm using the environment variable named CopyData here (which is set to $true), but I've also tried: "True", "true", $true", etc.

Each time, I get an error that states:

`Invoke-Expression: C:\Users\[redacted]\PowerShell\Modules\azure.datafactory.tools\0.96.0\private\ObjectProperty.ps1:22
Line |
  22 |      Invoke-Expression "$exp"
     |      ~~~~~~~~~~~~~~~~~~~~~~~~
     | The term 'True' is not recognized as a name of a cmdlet, function, script file, or executable program.
     | Check the spelling of the name, or if a path was included, verify that the path is correct and try again.`

In that script, it is executing an if/elseif/else statement based on the type of the parameter.  There is a code path for a string, for a PSCustomObject, and a else path that assumes that the value is numeric.  But no code path for boolean parameters.

What is the correct way to set a boolean parameter when using a staging .csv file?

**Comment by NowinskiK** [comment]
This is fixed now - see #203
